src/main/webapp/resources/js/components/project-user-groups/AddGroupButton.jsx

```diff
@@ -77,87 +79,84 @@ export function AddMembersButton() {
     visible,
   });
 
-  /*
-  Watch for changes to the forms visibility, when it becomes visible
-  set keyboard focus onto the user name input.
-   */
-  useEffect(() => {
-    if (visible) {
-      setTimeout(() => userRef.current.focus(), 100);
-    }
-  }, [visible]);
-
   /*
   Watch for changes to the debounced entered value for the user search.
   Once it changes send a request for filtered users.
    */
   useEffect(() => {
     if (debouncedQuery) {
-      getAvailableUsersForProject(debouncedQuery).then((data) =>
+      getAvailableGroupsForProject(debouncedQuery).then((data) =>
         setResults(data)
       );
     } else {
       setResults([]);
     }
   }, [debouncedQuery]);
 
+  /*
+  Watch for changes to the forms visibility, when it becomes visible
+  set keyboard focus onto the user name input.
+   */
+  useEffect(() => {
+    if (visible) {
+      setTimeout(() => groupRef.current.focus(), 100);
+    }
+  }, [visible]);
+
+  /*
+  Close the modal and reset the fields
+   */
+  const onCancel = () => {
+    form.resetFields();
+    setVisible(false);
+  };
+
+  /*
+  Add the user group
+   */
+  const addUserGroup = () => {
+    addUserGroupToProject({ groupId, role }).then((message) => {
+      onGroupAdded();
+      notification.success({ message });
+      form.resetFields();
+      setVisible(false);
+    });
+  };
+
   /*
   Before rendering format the results into Select.Option
    */
   const options = results.map((u) => (
     <Option className="t-new-member" key={u.identifier}>
       <Text style={{ marginRight: SPACE_XS }}>{u.label}</Text>
-      <Text type="secondary">{u.username}</Text>
     </Option>
   ));
 
-  /**
-   * Called when the user clicks the ok button.
-   */
-  const addUserToProject = () => {
-    addMemberToProject({ id: userId, role })
-      .then((message) => {
-        updateTable();
-        notification.success({ message });
-        setVisible(false);
-      })
-      .catch((message) => notification.error({ message }));
-  };
-
   return (
-    <>
-      <Button className="t-add-member-btn" onClick={() => setVisible(true)}>
-        {i18n("AddMemberButton.label")}
-      </Button>
-      <Modal
-        className="t-add-member-modal"
-        visible={visible}
-        okButtonProps={{ disabled: typeof userId === "undefined" }}
-        onCancel={() => setVisible(false)}
-        title={i18n("AddMemberButton.modal.title")}
-        onOk={addUserToProject}
-        okText={i18n("AddMemberButton.modal.okText")}
-      >
-        <Form layout="vertical" form={form} initialValues={{ role }}>
+    <Button onClick={() => setVisible(true)}>
+      {i18n("AddGroupButton.label")}
+      <Modal onCancel={onCancel} visible={visible} onOk={addUserGroup}
```

Comment:
Clicking the `cancel` or `x` button in the modal results in the modal just refreshing and not actually closing

Reply:
Fixed in a86d8c74626e724e9e35d6764848d2d291b03159